sdk/inc/azure/core/az_json.h

```diff
@@ -614,6 +614,11 @@ typedef struct
   /// and it shouldn't be modified by the caller.
   az_json_token token;
 
+  /// The depth of the current token. This read-only field tracks the recursive depth of the nested
+  /// objects or arrays within the JSON text processed so far, and it shouldn't be modified by the
+  /// caller.
+  int32_t current_depth;
```

Comment:
@JeffreyRichter, do you think it makes more sense to expose this field from the `az_json_token` struct rather than from the `az_json_reader`? I have it from the reader because it is easier to access from the caller (the user doesn't have to fish it out from the nested token field), but the depth can be thought of as a property of a particular JSON token.

Reply:
Can you share the customer code that uses this field.  The usage patterns would probably highlight if it should be on the reader on the token.

Reply:
I did above in the PR description.

Reply:
The reason why I didn't want to add it to the token is because right now, the token can be created as a standalone type, and used outside of the reader (at least as looking at the public fields). Adding depth to it turns it into something that has to be a part of a larger JSON payload.

Reply:
All that said, this is still in beta and we can revisit it separately, if necessary.

Reply:
fwiw I'd prefer in the json reader rather than the token unless there's a good reason for it to be in the token.

Reply:
Will close on it later, and decide if a change is needed or we are happy where we are:
https://github.com/Azure/azure-sdk-for-c/issues/1808